packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -43,6 +43,7 @@ const devHtmlHook: IndexHtmlTransformHook = async (
   html,
   { path: htmlPath, server }
 ) => {
+  // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
   const config = server?.config!
```

Comment:
This seems to be a design issue, cause optional chaining and non-null-assertion cannot be used on the same assignment